reference-implementation/lib/transform-stream.js

```diff
@@ -354,17 +359,20 @@ module.exports = class TransformStream {
       startPromise_resolve = resolve;
     });
 
-    const sink = new TransformStreamSink(this, startPromise);
-
-    this._writable = new WritableStream(sink, transformer.writableStrategy);
-
     const source = new TransformStreamSource(this, startPromise);
 
     this._readable = new ReadableStream(source, transformer.readableStrategy);
 
+    const sink = new TransformStreamSink(this, startPromise);
+
+    this._writable = new WritableStream(sink, transformer.writableStrategy);
+
     assert(this._writableController !== undefined);
     assert(this._readableController !== undefined);
 
+    const desiredSize = this._readableController.desiredSize;
+    TransformStreamSetBackpressure(this, desiredSize <= 0);
```

Comment:
Once the TransformStream constructor completes, user can start read()-ing from the readable side. If the strategy is set to HWM=0, i.e. configured to only pull() only when there's one or more read()s, pull() invocation is the only signal to remove backpressure. So, if you start with calling TransformStreamSetBackpressure(this, true), and TransformStreamSource() starts with being blocked on _backpressureChangePromise's fulfillment, I thought that there's no one who can fulfill it as pull() is suppressed.